File: java/src/main/java/io/fivexer/sdk/model/WorkerTimeseriesResult.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * One worker's bucketed history — the per-worker twin of the workspace timeseries.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerTimeseriesResult {
    private String workerId;
    private String from;
    private String to;
    private String bucket;
    private List<WorkerTimeseriesBucket> buckets;

    public String getWorkerId() { return workerId; }
    public String getFrom() { return from; }
    public String getTo() { return to; }
    public String getBucket() { return bucket; }
    public List<WorkerTimeseriesBucket> getBuckets() { return buckets; }
}

File: php/src/Model/WorkerTimeseriesResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * One worker's bucketed history — the per-worker twin of the workspace timeseries.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerTimeseriesResult
{
    public function __construct(
        public readonly string $workerId,
        public readonly string $from,
        public readonly string $to,
        public readonly string $bucket,
        /** @var list<WorkerTimeseriesBucket> */
        public readonly array $buckets,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            from: (string) ($data['from'] ?? ''),
            to: (string) ($data['to'] ?? ''),
            bucket: (string) ($data['bucket'] ?? ''),
            buckets: Json::parseEach($data, 'buckets', [WorkerTimeseriesBucket::class, 'fromArray']),
        );
    }
}
